find_email_addresses.py

```diff
@@ -0,0 +1,138 @@
+import argparse
+import re
+import tldextract
+from urlparse import urlparse 
+from twisted.internet import reactor, task
+from twisted.web.client import getPage
+from bs4 import BeautifulSoup
+
+# URLs already visited  
+already_visited = set()
+
+# Emails recognized so far
+emails_extracted = set()
+
+# Home page that we're scraping, including protocol and tld, used to compose relative links
+base_url = None
+
+# Domain name that needs to match the links to prevent the crawler from escaping 
+domain_name = None
+
+# Number of requests Twisted is currently waiting on 
+deferred_count = 0
+    
+# Max number of active connections 
+# Set this through trial and error - some sites will throttle you 
+# Your OS may also get upset about too much IO, so if you get weird errors try lowering this first
+concurrent_connection_limit = 10 
+
+# Queue for urls when the active connection limit is reached
+to_scrape_queue = []
+
+def main():
+    global deferred_count
+    global base_url
+    global domain_name
+    global to_scrape_queue
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('domain')
+    args = parser.parse_args()
+    
+    # Just so this works with the jana.com example... 
+    dom = args.domain
+    if len(dom) < 7 or 'http' not in dom[:4]:
+        dom = 'http://'+dom
+
+    to_scrape_queue.append(dom)
+
+    base_url = args.domain 
+    domain_name = tldextract.extract(args.domain).domain 
```

Comment:
Yeah, looking back at the language I think I was misinterpreting the subdomain restriction to mean "ignore all subdomains". I'll change it work on only the given subdomain (if given).